src/components/fabSpeedDial/fabSpeedDial.js

```diff
@@ -0,0 +1,130 @@
+   *   </md-fab-trigger>
+   *
+   *   <md-fab-actions>
+   *     <md-button><md-icon icon="/img/icons/user.svg"></md-icon></md-button>
+   *     <md-button><md-icon icon="/img/icons/group.svg"></md-icon></md-button>
+   *   </md-fab-actions>
+   * </md-fab-speed-dial>
+   * </hljs>
+   *
+   * @param {string=} md-direction From which direction you would like the speed dial to appear relative to the
+   * trigger element.
+   * @param {expression=} md-open Programmatically control whether or not the speed-dial is visible.
+   */
+  function MdFabSpeedDialDirective() {
+
+    return {
+      restrict: 'E',
+
+      scope: {
+        direction: '@?mdDirection',
+        isOpen: '=?mdOpen'
+      },
+
+      controller: FabSpeedDialController,
+
+      compile: function($element) {
+        // Wrap every child in a new div and add a class that we can scale independently
+        var children = $element.find('md-fab-actions').children(),
+          child, i;
+
+        for (i = 0; i < children.length; i++) {
+          child = children.eq(i);
+
+          child.wrap('<div class="md-fab-action-item">');
+        }
+
+        // Return our link function
+        return {
+          post: link
+        }
+      }
+    };
+
+    function FabSpeedDialController($scope, $element) {
+      var vm = this;
+
+      vm.scope = $scope;
+      vm.element = $element;
+
+      // Set the default direction to 'down' if none is specified
+      if (typeof $scope.direction === 'undefined') {
+        $scope.$apply('direction = "down"');
+      }
+
+      // Set the default to be closed
+      if (typeof $scope.isOpen === 'undefined') {
+        $scope.$apply('isOpen = false');
+      }
+
+      vm.focus = function() {
+        vm.scope.$apply('isOpen = true');
+      };
+
+      vm.blur = function() {
+        vm.scope.$apply('isOpen = false');
+      };
+
+    }
+
+    function link($scope, $element, $attrs, ctrl) {
+      var i, trigger, child, children, focusable;
+
+      // Add the focus/blur events to the trigger element
+      trigger = $element.find('md-fab-trigger').children().eq(0);
+
+      trigger.on('focus', ctrl.focus);
+      trigger.on('blur', ctrl.blur);
+
+      // Add some focus/blur events to whatever (post-compiled) elements the user added in the speed dial
+      children = $element.find('md-fab-actions').children();
+
+      for (i = 0; i < children.length; i++) {
+        focusable = children.eq(i).children().eq(0);
+
+        focusable.on('focus', ctrl.focus);
+        focusable.on('blur', ctrl.blur);
+      }
+
+      $element.on('mouseenter', ctrl.focus);
+      $element.on('mouseleave', ctrl.blur);
+
+      // Watch for changes to the direction and update classes/attributes
+      $scope.$watch('direction', function(newDir, oldDir) {
+        // Add the appropriate classes so we can target the direction in the CSS
+        $element.removeClass('md-' + oldDir);
+        $element.addClass('md-' + newDir);
```

Comment:
Maybe a little far-fetched, but using `$animate.add/removeClass()` would allow developers to set up nice transition animations. (Not that anyone is ever likely to do so...)